src/IceRpc/IceRpcTypeExtensions.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) ZeroC, Inc.
+
+using ZeroC.Slice;
+
+namespace IceRpc;
+
+/// <summary>Provides extension methods for <see cref="Type" />.</summary>
+public static class TypeExtensions
+{
+    /// <summary>Retrieves the default service path with the attribute <see cref="DefaultServicePathAttribute"/>.
+    /// </summary>
+    /// <param name="type">The type of an interface.</param>
+    /// <returns>The default service path.</returns>
+    /// <exception cref="ArgumentException">Thrown if <paramref name="type" /> is not an interface, or if
+    /// it does not have a <see cref="DefaultServicePathAttribute" /> attribute.</exception>
+    public static string GetDefaultServicePath(this Type type)
+    {
+        if (type.IsInterface)
+        {
+            object[] attributes = type.GetCustomAttributes(typeof(DefaultServicePathAttribute), false);
+            if (attributes.Length == 1 && attributes[0] is DefaultServicePathAttribute defaultServicePath)
+            {
+                return defaultServicePath.Value;
+            }
+            else
+            {
+                throw new ArgumentException(
```

Comment:
If you accidentally have two `[IceRpc.DefaultServicePath(""/{service.Name}"")]` on the same interface by mistake then `attributes.Length` == 1 would be false and we would get into the error case here. The error message then would not be correct as it does have the `DefaultServicePathAttribute ` attribute, there are just multiple.

This would probably be super uncommon so it might not be worth handling with a seperate error.

Reply:
The compiler handles this case, the default is AllowMultiple = false

See https://learn.microsoft.com/en-us/dotnet/api/system.attributeusageattribute.allowmultiple?view=net-7.0